src/geometry/euclidean/circle.lean

```diff
@@ -0,0 +1,147 @@
+Authors: Manuel Candales
+-/
+import geometry.euclidean.basic
+
+/-!
+# Circles
+
+This file proves basic geometrical results about distances and angles
+in circles in real inner product spaces and Euclidean affine spaces.
+-/
+
+open_locale euclidean_geometry real_inner_product_space real
+
+variables {V : Type*} [inner_product_space ℝ V]
+
+namespace inner_product_geometry
+/-!
+### Geometrical results on circles in real inner product spaces
+
+This section develops some results on circles in real inner product spaces,
+which are used to deduce corresponding results for Euclidean affine spaces.
+-/
+
+lemma mul_norm_eq_sub_norm_pow_two {x y z : V} (h₁ : ∃ k : ℝ, k ≠ 1 ∧ x + y = k • (x - y))
+  (h₂ : ∥z - y∥ = ∥z + y∥) : ∥x - y∥ * ∥x + y∥ = abs (∥z + y∥ ^ 2 - ∥z - x∥ ^ 2) :=
+begin
+  obtain ⟨k, hk_ne_one, hk⟩ := h₁,
+  let r := (k - 1)⁻¹ * (k + 1),
+
+  have hxy : x = r • y,
+  { rw [← smul_smul, eq_inv_smul_iff' (sub_ne_zero.mpr hk_ne_one), ← sub_eq_zero],
+    calc  (k - 1) • x - (k + 1) • y
+        = (k • x - x) - (k • y + y) : by simp_rw [sub_smul, add_smul, one_smul]
+    ... = (k • x - k • y) - (x + y) : by simp_rw [← sub_sub, sub_right_comm]
+    ... = k • (x - y) - (x + y)     : by rw ← smul_sub k x y
+    ... = 0                         : sub_eq_zero.mpr hk.symm },
+
+  have hzy : ⟪z, y⟫ = 0,
+  { rw [← eq_of_pow_two_eq_pow_two (norm_nonneg (z - y)) (norm_nonneg (z + y)),
+        norm_add_pow_two_real, norm_sub_pow_two_real] at h₂,
+    linarith },
+
+  have hzx : ⟪z, x⟫ = 0 := by rw [hxy, inner_smul_right, hzy, mul_zero],
+
+  symmetry,
+
+  calc  abs (∥z + y∥ ^ 2 - ∥z - x∥ ^ 2)
+      = abs (∥x∥ ^ 2 - ∥y∥ ^ 2)            : by simp [norm_add_pow_two_real, norm_sub_pow_two_real,
+                                                    hzy, hzx, abs_sub]
+  ... = abs (∥r∥ ^ 2 * ∥y∥ ^ 2 - ∥y∥ ^ 2)   : by rw [hxy, norm_smul, mul_pow]
+  ... = abs (r ^ 2 * ∥y∥ ^ 2 - ∥y∥ ^ 2)    : by rw [real.norm_eq_abs, sqr_abs]
+  ... = abs ((r - 1) * (r + 1) * ∥y∥ ^ 2) : by ring_nf
+  ... = ∥r - 1∥ * ∥r + 1∥ * ∥y∥ ^ 2         : by simp [abs_mul, real.norm_eq_abs]
+  ... = ∥r - 1∥ * ∥y∥ * (∥r + 1∥ * ∥y∥)      : by ring
+  ... = ∥(r - 1) • y∥ * ∥(r + 1) • y∥      : by simp_rw [norm_smul]
+  ... = ∥x - y∥ * ∥x + y∥                  : by simp [sub_smul, add_smul, hxy],
+end
+
+end inner_product_geometry
+
+namespace euclidean_geometry
+/-!
+### Geometrical results on circles in Euclidean affine spaces
+
+This section develops some results on circles in Euclidean affine spaces.
+-/
+
+open inner_product_geometry
+
+variables {P : Type*} [metric_space P] [normed_add_torsor V P]
+include V
+
+/-- If `P` is a point on the line `AB` and `Q` is equidistant from `A` and `B`, then
+`AP * BP = abs (BQ ^ 2 - PQ ^ 2)`. -/
+lemma mul_dist_eq_sub_dist_pow_two {a b p q : P} (hp : ∃ k : ℝ, k ≠ 1 ∧ b -ᵥ p = k • (a -ᵥ p))
+  (hq : dist a q = dist b q) : (dist a p) * (dist b p) = abs ((dist b q) ^ 2 - (dist p q) ^ 2) :=
```

Comment:
```suggestion
  (hq : dist a q = dist b q) : dist a p * dist b p = abs (dist b q ^ 2 - dist p q ^ 2) :=
```
You don't need parentheses around `dist _ _` (you should make this change through the rest of the file as well).